app/state_machines/configurable_state_machine/machine.rb

```diff
@@ -118,7 +118,7 @@ def trigger_event(event:, params:)
           to_state = find_destination_state(event_config: event_config, user: params[:acting_user])
           to_workflow = find_destination_workflow(event_config: event_config)
           CaseTransition.unset_most_recent(@kase)
-          write_transition(event: event, to_state: to_state, to_workflow: to_workflow, params: params)
+          write_transition(event: event, to_state: to_state, to_workflow: to_workflow, params: params.clone)
```

Comment:
Having to add `.clone` here seems like a smell. `write_transition` should be handling this particular detail, either by not deleting keys from `params` or by cloning it. 🚨

Reply:
yep, you're right.  Will fix